src/application-manager.coffee

```diff
@@ -909,11 +909,18 @@ module.exports = class ApplicationManager extends EventEmitter
 							if _.every(nextSteps, { action: 'noop' })
 								volumePromises.push(@removeAllVolumesForApp(checkInt(appId)))
 			newDownloads = _.filter(nextSteps, (s) -> s.action == 'fetch').length
-			if !ignoreImages and delta and newDownloads > 0
-				downloadsToBlock = downloading.length + newDownloads - constants.maxDeltaDownloads
-				while downloadsToBlock > 0
-					_.pull(nextSteps, _.find(nextSteps, action: 'fetch'))
-					downloadsToBlock -= 1
+
+			if !ignoreImages and delta and newDownloads > 0 and availableImages > 0
+				# Check that this is not the first pull for an
+				# application, as we want to download all images then
+				availableFromThisApp = _.filter availableImages, (img) ->
+					img.appId = appId
```

Comment:
You're doing an assignment here, when you really meant:

```suggestion
					img.appId is appId
```

Reply:
woah, good catch